### PR TITLE
improved prime numbers implementation

### PR DESCRIPTION
**Pull request by nikolasvargas**

### CODE REVIEW

File: maths/prime_numbers.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
-from typing import List
+from typing import Generator, List
 
 
-def primes(max: int) -> List[int]:
+def primes(max: int) -> Generator[int, None, None]:
     """
     Return a list of all primes numbers up to max.
     >>> primes(10)
```
Comment (cclauss) [inline maths/prime_numbers.py]: Please change this to __list(primes(10))__ so this doctest passes.

File: maths/prime_numbers.py [hunk 2]
```diff
@@ -13,16 +13,16 @@ def primes(max: int) -> List[int]:
     >>> primes(1_000_000)[-1]
```
Comment (cclauss) [inline maths/prime_numbers.py]: Please change this to __list(primes(1_000_000))[-1]__ so this doctest passes.

Please add doctests for __list(primes(0))__, __list(primes(-1))__, __list(primes(-10))__

You can try these on your own computer with __python3 -m doctest -v maths/prime_numbers.py__

File: maths/prime_numbers.py [hunk 3]
```diff
@@ -13,16 +13,16 @@ def primes(max: int) -> List[int]:
...
-    numbers = [False] * max
-    ret = []
-    for i in range(2, max):
-        if not numbers[i]:
-            for j in range(i, max, i):
-                numbers[j] = True
-            ret.append(i)
-    return ret
+    numbers: Generator = (i for i in range(1, (max + 1)))
+    for i in filter(lambda x: x > 1, numbers):
```
Comment (cclauss) [inline maths/prime_numbers.py]: __for i in [x for x in numbers if x > 1]:__

File: maths/prime_numbers.py [hunk 4]
```diff
@@ -13,16 +13,16 @@ def primes(max: int) -> List[int]:
...
+        for j in range(2, i):
+            if (i % j) == 0:
+                break
+        else:
+            yield i
 
 
 if __name__ == "__main__":
-    print(primes(int(input("Calculate primes up to:\n>> "))))
+    number = int(input("Calculate primes up to:\n>> "))
```
Comment (cclauss) [inline maths/prime_numbers.py]: As discussed in CONTRIBUTING.md, use __str.strip()__ to gracefully deal with leading or trailing whitespace in user input.  __number = int(input("Calculate primes up to:\n>> ").strip())__

File: maths/prime_numbers.py (hunk 3 above)
Comment (nikolasvargas) [reply maths/prime_numbers.py]: filter returns an iterator, better then a full list, i guess

File: maths/prime_numbers.py (hunk 4 above)
Comment (nikolasvargas) [reply maths/prime_numbers.py]: fixed

File: maths/prime_numbers.py (hunk 1 above)
Comment (nikolasvargas) [reply maths/prime_numbers.py]: fixed

File: maths/prime_numbers.py (hunk 3 above)
Comment (cclauss) [reply maths/prime_numbers.py]: __ for i in (x for x in numbers if x > 1):__

**Pull request merged**